docs/contributor/pocs/integrate-prometheus-with-telemetry-manager-using-alerting.md

```diff
@@ -0,0 +1,258 @@
+
+			// TODO: add more context about which objects have to reconciled 
+			reconcileTriggerChan <- event.GenericEvent{}
+
+			w.WriteHeader(http.StatusOK)
+		}
+		
+		mux := http.NewServeMux()
+		mux.HandleFunc("/api/v2/alerts", handler)
+
+		server := &http.Server{
+			Addr:              ":9090",
+			ReadHeaderTimeout: 10 * time.Second,
+			Handler:           mux,
+		}
+
+		if serverErr := server.ListenAndServe(); serverErr != nil {
+			mutex.Lock()
+			setupLog.Error(serverErr, "Cannot start webhook server")
+			mutex.Unlock()
+		}
+	}()
+```
+5. Trigger reconciliation in MetricPipelineController whenever the endpoint is called by Prometheus:
+```go
+func NewMetricPipelineReconciler(client client.Client, reconcileTriggerChan chan event.GenericEvent, reconciler *metricpipeline.Reconciler) *MetricPipelineReconciler {
+	return &MetricPipelineReconciler{
+		Client:     client,
+		reconciler: reconciler,
+		Client:      client,
+		reconciler:  reconciler,
+        reconcileTriggerChan: reconcileTriggerChan,
+	}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *MetricPipelineReconciler) SetupWithManager(mgr ctrl.Manager) error {
+    // We use `Watches` instead of `Owns` to trigger a reconciliation also when owned objects without the controller flag are changed.
+    return ctrl.NewControllerManagedBy(mgr).
+            For(&telemetryv1alpha1.MetricPipeline{}).
+            WatchesRawSource(&source.Channel{Source: r.reconcileTriggerChan},
+            handler.EnqueueRequestsFromMapFunc(r.mapPrometheusAlertEvent)).
+		...
+}
+
+func (r *MetricPipelineReconciler) mapPrometheusAlertEvent(ctx context.Context, _ client.Object) []reconcile.Request {
+    logf.FromContext(ctx).Info("Handling Prometheus alert event")
+    requests, err := r.createRequestsForAllPipelines(ctx)
+    if err != nil {
+    logf.FromContext(ctx).Error(err, "Unable to create reconcile requests")
+    }
+    return requests
+}
+```
+6. Query Prometheus alerts in the Reconcile function:
+```go
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/api"
+	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const prometheusAPIURL = "http://prometheus-server.default:80"
+
+func queryAlerts(ctx context.Context) error {
+	client, err := api.NewClient(api.Config{
+		Address: prometheusAPIURL,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create Prometheus client: %w", err)
+	}
+
+	v1api := promv1.NewAPI(client)
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	start := time.Now()
+	alerts, err := v1api.Alerts(ctx)
+
+	if err != nil {
+		return fmt.Errorf("failed to query Prometheus alerts: %w", err)
+	}
+
+	logf.FromContext(ctx).Info("Prometheus alert query succeeded!",
+		"elapsed_ms", time.Since(start).Milliseconds(),
+		"alerts", alerts)
+	return nil
+}
+```
+
+8. Add a Kubernetes service for the alerts endpoint to the kustomize file:
```

Comment:
where's step 7?

Anyway, 3 blank spaces needed

Reply:
My bad, fixed the list